### ISSUE TITLE
Handle networkx null and empty generators

### ISSUE THREAD

**Issue by pausz** [starsimhub/starsim#233]
Wondering if we could have a starsim "null" network, where null could be a an empty graph at the beginning of the simulation and edges are added at a later time; or, "null" could be a network with edge weights all set to zero (could be achieved with `ss.RandomNetwork(n_contacts=0)` after integrating [Dan's fix in RandomNetwork](https://github.com/amath-idm/starsim/commit/be492fc45e79868310a357221c13d25d987fa629#diff-f6546bd9b54a712bd805b634255db9840ffe81486c464e1010e6e420435a5695)) 

I think a null network could be useful for debugging purposes.  

Some findings:
##### Test code
```python
    ppl = ss.People(100)
    ppl.networks = ss.ndict(ss.static(graph=nx.null_graph))

    syph = ss.Syphilis()

    # Sim
    sim = ss.Sim(people=ppl, diseases=[syph,])
    sim.run()
```
##### Output:
```
Traceback (most recent call last):
  File "/home/paula/Projects/IDM/dev-starsim/starsim/tests/test_null_network.py", line 31, in <module>
    sim1 = test_null_network()
  File "/home/paula/Projects/IDM/dev-starsim/starsim/tests/test_null_network.py", line 23, in test_null_network
    sim.run()
  File "/home/paula/Projects/IDM/dev-starsim/starsim/starsim/sim.py", line 422, in run
    self.initialize()
  File "/home/paula/Projects/IDM/dev-starsim/starsim/starsim/sim.py", line 95, in initialize
    self.init_networks()
  File "/home/paula/Projects/IDM/dev-starsim/starsim/starsim/sim.py", line 270, in init_networks
    self.people.networks.initialize(self)
  File "/home/paula/Projects/IDM/dev-starsim/starsim/starsim/networks/networks.py", line 300, in initialize
    nw.initialize(sim)
  File "/home/paula/Projects/IDM/dev-starsim/starsim/starsim/networks/networks.py", line 912, in initialize
    self.graph = self.graph(n = popsize, **self.kwargs)
  File "/home/paula/miniconda3/envs/py310-starsim-dev/lib/python3.10/site-packages/networkx/utils/backends.py", line 412, in __call__
    return self.orig_func(*args, **kwargs)
TypeError: null_graph() got an unexpected keyword argument 'n'
```

which is expected because networkx null_graph does not take a size argument.

**Comment by cliffckerr** [comment]
This is partly fixed by #339

**Comment by cliffckerr** [comment]
(Didn't mean to close!)

**Comment by cliffckerr** [comment]
Closed by #369
